core/cont/inc/TCollectionProxyInfo.h

```diff
@@ -613,16 +631,18 @@ namespace Detail {
 
    };
 
-   template <> struct TCollectionProxyInfo::Type<std::vector<Bool_t> >
-   : public TCollectionProxyInfo::Address<std::vector<Bool_t>::const_reference>
+   // This specialization is chosen if T is a vector<bool, A>, irrespective of the nature
+   // of the allocator A represents.
+   template <class T> struct TCollectionProxyInfo::Type<T, VectorBoolTrue>
+   : public TCollectionProxyInfo::Address<typename T::const_reference>
```

Comment:
would something like this simplify the `VectorBoolTrue/False` logic? https://godbolt.org/z/_glVS7